xc-tool/build.rs

```diff
@@ -0,0 +1,107 @@
+use std::{env, fs, path::PathBuf, path::Path};
+
+use ethers::contract::Abigen;
+use ethers::solc::{remappings::Remapping, Artifact, Project, ProjectPathsConfig, Solc};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Can we split this up into runtime-specific methods? Would be good to have all the EVM build stuff in one place, all the Solana stuff in another, etc.

Reply:
It would also be good to put the EVM binding code in a function called something like `generate_evm_bindings`, so it's clear what it does.

Reply:
That's a good idea. If we get a fix for the codegen, I'll do that.